examples/plot_metricframe_beyond_binary_classification.py

```diff
@@ -0,0 +1,249 @@
+s_f = [chr(ord('a')+x) for x in temp]
+
+# %%
+# To use :func:`~sklearn.metrics.confusion_matrix`, we
+# need to prebind the `labels` argument, since it is possible
+# that some of the subgroups will not contain all of
+# the possible labels
+
+
+conf_mat = functools.partial(skm.confusion_matrix,
+                             labels=np.unique(y_true))
+
+# %%
+# With this now available, we can create our
+# :class:`~fairlearn.metrics.MetricFrame`:
+
+mf = MetricFrame(metrics={'conf_mat': conf_mat},
+                 y_true=y_true,
+                 y_pred=y_pred,
+                 sensitive_features=s_f)
+
+# %%
+# From this, we can view the overall confusion matrix:
+
+mf.overall
+
+# %%
+# And also the confusion matrices for each subgroup:
+
+mf.by_group
+
+# %%
+# Obviously, the other methods such as
+# :meth:`~fairlearn.metrics.MetricFrame.group_min`
+# will not work, since operations such as 'less than'
+# are not well defined for matrices.
+
+# %%
+# Metric functions with different return types can also
+# be mixed in a single :class:`~fairlearn.metrics.MetricFrame`.
+# For example:
+
+recall = functools.partial(skm.recall_score, average='macro')
+
+mf2 = MetricFrame(metrics={'conf_mat': conf_mat,
+                           'recall': recall
+                           },
+                  y_true=y_true,
+                  y_pred=y_pred,
+                  sensitive_features=s_f)
+
+print("Overall values")
+print(mf2.overall)
+print("Values by group")
+print(mf2.by_group)
+
+
+# %%
+# Non-scalar Inputs
+# =================
+#
+# :class:`~fairlearn.metrics.MetricFrame` does not require
+# its inputs to be scalars either. To demonstrate this, we
+# will use an image recognition example (kindly supplied by
+# Ferdane Bekmezci, Hamid Vaezi Joze and Samira Pouyanfar).
+#
+# Image recognition algorithms frequently construct a bounding
+# box around regions where they have found their target features.
+# For example, if an algorithm detects a face in an image, it
+# will place a bounding box around it. These bounding boxes
+# constitute `y_pred` for :class:`~fairlearn.metrics.MetricFrame`.
+# The `y_true` values then come from bounding boxes marked by
+# human labellers.
+#
+# Bounding boxes are often compared using the 'iou' metric.
+# This computes the intersection and the union of the two
+# bounding boxes, and returns the ratio of their areas.
+# If the bounding boxes are identical, then the metric will
+# be 1; if disjoint then it will be 0. A function to do this is:
+
+def bounding_box_iou(box_A_input, box_B_input):
+    # The inputs are array-likes in the form
+    # [x_0, y_0, delta_x,delta_y]
+    # where the deltas are positive
+
+    box_A = np.array(box_A_input)
+    box_B = np.array(box_B_input)
+
+    assert box_A[2] >= 0, "Bad delta_x for box_A"
+    assert box_A[3] >= 0, "Bad delta y for box_A"
+    assert box_B[2] >= 0, "Bad delta x for box_B"
+    assert box_B[3] >= 0, "Bad delta y for box_B"
```

Comment:
we should raise to validate input instead of assert

Reply:
OK